internal/transport/http2_server.go

```diff
@@ -389,6 +390,41 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			if timeout, err = decodeTimeout(hf.Value); err != nil {
 				headerError = true
 			}
+		// "Transports must consider requests containing the Connection header
+		// as malformed." - A41
+		case "connection":
+			t.controlBuf.put(&cleanupStream{
+				streamID: streamID,
+				rst:      true,
+				rstCode:  http2.ErrCodeProtocol,
+				onWrite:  func() {},
+			})
+			return false
+		// "If multiple Host headers or multiple :authority headers are present,
+		// the request must be rejected by RST_STREAM with HTTP/2 error code
+		// PROTOCOL_ERROR." - A41
+		case ":authority":
```

Comment:
Does this actually remove `:authority` from the returned metadata?
Without this, it's handled in `default:`

Reply:
And similarly, `:path` is not added to metadata I think.

Reply:
Ah, lol, great catch. I told ya I can't figure out how to test this, maybe I would've caught this obvious error lol.

Reply:
Btw I got Doug's help on how to test this in our 1:1, working on that now.

Reply:
Are you going to add the tests to this PR?

Reply:
And `:path: is still not in mdata I believe.
Though that might not necessary? The interceptors already have the path, just that they are in the `Info` struct, not metadata

Reply:
Yeah, I'll add the test to this PR. Yeah, you're right, it's not neccesary from my interpretation. Path matcher maps to the full method name the RPC is calling on the server side (https://github.com/grpc/grpc-go/blob/master/internal/xds/rbac/matchers.go#L291). However, A41 states this: "The header field is not entirely 1:1 with gRPC Metadata, in part because which HTTP headers are present in Metadata is not 100% consistent cross-language. For this design, headers can include :method, :authority, and :path matchers and they should match the values received on-the-wire independent of whether they are stored in Metadata or in separate APIs", so perhaps I need to add it.